apps/api/lib/api/client/channel.ex

```diff
@@ -1,20 +1,123 @@
 defmodule API.Client.Channel do
   use API, :channel
-  alias Domain.Clients
+  alias API.Client.Views
+  alias Domain.{Clients, Resources, Gateways, Relays}
 
   # TODO: we need to self-terminate channel once the user token is set to expire, preventing
   # users from holding infinite session for if they want to keep websocket open for a while
 
   @impl true
   def join("client", _payload, socket) do
+    expires_in = DateTime.diff(socket.assigns.expires_at, DateTime.utc_now(), :millisecond)
+    Process.send_after(self(), :token_expired, expires_in)
```

Comment:
This should be a pretty rare case because socket verifies expiration time when you connect, so the only way to hit this is if it expired between two functions calls. Anyway, I'll add a check for this.